Quaver.API/Maps/Qua.cs

```diff
@@ -176,7 +173,7 @@ public class Qua
         /// </summary>
         /// <returns></returns>
         [YamlIgnore]
-        public int Length => HitObjects.Count == 0 ? 0 : HitObjects.Max(x => Math.Max(x.StartTime, x.EndTime));
+        public int Length => HitObjects.Count == 0 ? 0 : Math.Max(HitObjects[^1].StartTime, HitObjects[^1].EndTime);
```

Comment:
HitObjects are only sorted by StartTime. There is no guarantee that the last hitobject has the greatest EndTime. Consider the second last note being longer than the last and released later than the last.

Reply:
That's a good catch, thanks.